packages/firestore/lite/src/api/reference.ts

```diff
@@ -49,3 +55,222 @@ export class DocumentReference<T = firestore.DocumentData>
+  constructor(
+    readonly firestore: Firestore,
+    readonly _query: InternalQuery,
+    readonly _converter?: FirestoreDataConverter<T>
+  ) {}
+
+  where(
+    fieldPath: string | firestore.FieldPath,
+    opStr: firestore.WhereFilterOp,
+    value: unknown
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  orderBy(
+    fieldPath: string | firestore.FieldPath,
+    directionStr?: firestore.OrderByDirection
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  limit(limit: number): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  limitToLast(limit: number): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  startAfter(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  startAt(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  endAt(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  endBefore(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  withConverter<U>(
+    converter: firestore.FirestoreDataConverter<U>
+  ): firestore.Query<U> {
+    return new Query<U>(this.firestore, this._query, converter);
+  }
+}
+
+export class CollectionReference<T = firestore.DocumentData> extends Query<T>
+  implements firestore.CollectionReference<T> {
+  constructor(
+    readonly firestore: Firestore,
+    readonly _path: ResourcePath,
+    readonly _converter?: firestore.FirestoreDataConverter<T>
+  ) {
+    super(firestore, InternalQuery.atPath(_path), _converter);
+  }
+
+  get id(): string {
+    return this._query.path.lastSegment();
+  }
+
+  get path(): string {
+    return this._query.path.canonicalString();
+  }
+
+  withConverter<U>(
+    converter: firestore.FirestoreDataConverter<U>
+  ): firestore.CollectionReference<U> {
+    return new CollectionReference<U>(this.firestore, this._path, converter);
+  }
+}
+
+export function collection(
+  firestore: FirebaseFirestore,
+  collectionPath: string
+): CollectionReference<firestore.DocumentData>;
+export function collection(
+  reference: DocumentReference,
+  collectionPath: string
+): CollectionReference<firestore.DocumentData>;
+export function collection(
```

Comment:
Combines https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/api/database.ts;l=561 and https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/api/database.ts;l=981